Remote updates with ansible-pull

jasoncorlett opened the pull request:
**Changes:**
* Paths in ansible/nimbuserver.yml now relative, allows updating with ansible-pull
* New nimbus-server-update script to run ansible-pull easily
* Tweaks to nimbusserver.json so ansible will work with relative paths, and make using a new CentOS iso image easier

**To test:**
```
ansible-pull -U git://github.com/jasoncorlett/nimbusserver-ansible -C remote-update ansible/nimbusserver.yml
```
OR, if the script is installed:
```
export NIMBUS_UPDATE_URL="git://github.com/jasoncorlett/nimbusserver-ansible"
export NIMBUS_UPDATE_BRANCH="remote-update"
nimbus-server-update
```

jasoncorlett committed:
Use relative paths for all provided files

This allows ansible-pull to clone the repo in another location

jasoncorlett committed:
Use tag_name instead of name when identifying releases

Name may not always match the version number

jasoncorlett committed:
Specify playbook_dir, configurable iso properties

- playbook_dir transfers entire directory without separate stpes,
  allowing use of relative paths in ansible configuration (important
  for updating later)
- configurable iso properties using variables allow for easier updating
  of the base system and specifying where the file exists on the host

jasoncorlett committed:
Specify a docker-app tag to use

jasoncorlett committed:
Do not disconnect existing containers from network

jasoncorlett committed:
Remove python-gssapi to prevent errors

The docker_network module fails when an incorrect version of
python-gssapi is present.

jasoncorlett committed:
Add new script to perform ansible-pull

This script can be used to run the latest playbook from github

jasoncorlett committed:
Merge branch 'remote-update' of git+ssh://192.168.172.128:/home/demo/IdeaProjects_docker/nimbusserver-ansible into remote-update

jasoncorlett committed:
Correct URL variable name and ansible command

jasoncorlett committed:
Updated changelog

jhrabi commented on changelog.md:
```diff
@@ -1,6 +1,14 @@
 # nimbusserver-ansible
 All notable changes to this project will be documented in this file. The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 
+# [2019.04] - 2019-04-09
```
I'm looking at making the naming convention Year+R(number). This way it doesn't matter when we release it,
I'm going to confirm this with Keith.

jhrabi reviewed (commented):
We need to update quickstart.html to add the new version as well.

jasoncorlett committed:
Updated to use YYYY.RN version scheme

jasoncorlett committed:
Dynamic version in quickstart.html file

jasoncorlett committed:
Merge branch 'remote-update' of git+ssh://192.168.172.128:/home/demo/IdeaProjects_docker/nimbusserver-ansible into remote-update

jasoncorlett committed:
Version in quickstart set dynamically

Use the packer "version" variable to fill out the file.
Switched versioning scheme in changelog.

jhrabi reviewed (approved):
LGTM

jasoncorlett committed:
68038 - Update bookmarks

jasoncorlett committed:
Updated chrome bookmarks

jhrabi reviewed (approved):
LGTM

jasoncorlett committed:
Use the run date when no version is provided

jhrabi reviewed (approved):
LGTM

jasoncorlett committed:
Delete user config to force bookmark import

jhrabi commented on ansible/nimbusserver.yml:
```diff
@@ -221,6 +221,10 @@
       replace: 'MaxSessions=1'
       backup: yes
+
+  - name: Remove existing chrome directory
```
Isn't this redundant with the change made below for the bookmarks?

jasoncorlett replied:
Indeed, forgot I already added this. I'll remove this one with my next push.

jasoncorlett committed:
Remove redundant step

jasoncorlett committed:
Remove testing tag

jasoncorlett committed:
Updating firefox bookmarks

jhrabi reviewed (approved):
LGTM
